spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/feign/hystrix/HystrixRequestContextEnablerFilter.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2013-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.netflix.feign.hystrix;
+
+import java.io.IOException;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.netflix.hystrix.strategy.concurrency.HystrixRequestContext;
+
+/**
+ * @author Daniel Lavoie
+ */
+public class HystrixRequestContextEnablerFilter implements Filter {
+	private static Logger LOGGER = LoggerFactory
+			.getLogger(HystrixRequestContextEnablerFilter.class);
+
+	@Override
+	public void doFilter(ServletRequest request, ServletResponse response,
+			FilterChain chain) throws IOException, ServletException {
+		if (LOGGER.isTraceEnabled())
+			LOGGER.trace("Initializing Hystrix Request Context.");
+
+		HystrixRequestContext context = HystrixRequestContext.initializeContext();
+		try {
+			chain.doFilter(request, response);
```

Comment:
I'll just quote the excellent Juan Naverrete [post](https://jfconavarrete.wordpress.com/2014/09/15/make-spring-security-context-available-inside-a-hystrix-command/). Basically, it initialize the holder which allow the SecurityContext to be request scoped rather than thread scoped. It's the little box where you'll store the SecurityContext.

> The HystrixRequestContext must be initialized at the beginning of each request before RequestVariable functionality can be used. This enables the HystrixRequestContext to contain the state and manage the lifecycle of HystrixRequestVariableDefault objects that provide request scoped (rather than only thread scoped) variables so that multiple threads within a single request can share state.